### PR TITLE
fix IPC::waitUntilAccepted

### LINKED ISSUE #2360: Can't start a second instance
@tux3 Why waiting less make qTox not able to start a second instance? 59e06920692a5966645bf60baa8753f440842651

Is the timeout too low?
https://github.com/tux3/qTox/blob/59e06920692a5966645bf60baa8753f440842651/src/ipc.cpp#L192

### PR DESCRIPTION
**Pull request by antis81**
Fixes #2360.

### CODE REVIEW

**Comment by antis81** [comment]
Hm… @tux3 could you please help in finding the reason for #2360? However the sleep halts qTox for 10 secs. on startup and is definitely **not** the way how things should be done.

**Comment by antis81** [comment]
@tux3 Found it. Will add a fix here.

**Comment by antis81** [comment]
@tux3 Can you please review/verify this?

File: src/main.cpp [hunk 1]
```diff
@@ -233,15 +232,10 @@ int main(int argc, char *argv[])
...
-            dest = Settings::getInstance().getCurrentProfileId();
-
-        time_t event = ipc.postEvent("activate", QByteArray(), dest);
-        if (ipc.waitUntilAccepted(event, 2))
+        time_t event = ipc.postEvent("activate");
+        if (!ipc.waitUntilAccepted(event, 2))
         {
-            if (!ipc.isCurrentOwner())
-                return EXIT_SUCCESS;
+            return EXIT_SUCCESS;
```
Comment (antis81) [inline src/main.cpp:238]: We probably should return error code, shouldn't we?

File: src/ipc.cpp [hunk 2]
```diff
@@ -188,7 +189,6 @@ bool IPC::waitUntilAccepted(time_t postTime, int32_t timeout/*=-1*/)
     while (!(result = isEventAccepted(postTime)))
     {
         qApp->processEvents();
-        QThread::msleep(10);
```
Comment (tux3) [inline src/ipc.cpp:191]: Isn't it going to spin at 100% CPU if you remove the sleep?

Comment (antis81) [inline src/ipc.cpp:191]: No, `qApp->processEvents();` prevents it. Rather than that, qTox starts immediately the second you open it… :)

Comment (antis81) [inline src/ipc.cpp:191]: Sorry, I was wrong -> CPU actually spins to 100%. No problem though - I'll fix it and after we should be ok here.

**Comment by antis81** [comment]
@tux3 Should be good now. You can check by setting the timeout to -1 (infinite) and never set result == true.

**Comment by antis81** [comment]
@tux3 Should I merge this?

**Comment by tux3** [comment]
@antis81 Yes, it looks good to me.

**Pull request merged**